crates/re_examples/src/lib.rs

```diff
@@ -0,0 +1,171 @@
+                    eprintln!("Skipping example {name:?} because it is not included in the {self} channel");
+                    continue;
+                }
+
+                eprintln!("Adding example {name:?}");
+                examples.push(Example {
+                    name,
+                    title: readme.title,
+                    description: readme.description,
+                    tags: readme.tags,
+                    thumbnail_url: readme.thumbnail,
+                    thumbnail_dimensions: readme.thumbnail_dimensions,
+                    script_path: folder.path().join("main.py"),
+                    script_args: readme.build_args,
+                    kind,
+                });
+            }
+        }
+
+        if examples.is_empty() {
+            anyhow::bail!("No examples found in {}", dir.display())
+        }
+
+        examples.sort_unstable_by(|a, b| a.name.cmp(&b.name));
+        Ok(examples)
+    }
+}
+
+impl Display for Channel {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let s = match self {
+            Channel::Main => "main",
+            Channel::Nightly => "nightly",
+        };
+        f.write_str(s)
+    }
+}
+
+impl FromStr for Channel {
+    type Err = ();
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "main" => Ok(Self::Main),
+            "nightly" => Ok(Self::Nightly),
+            _ => Err(()),
+        }
+    }
+}
+
+#[derive(Clone, Copy)]
+pub enum ExampleKind {
+    Ignore,
+    Demo,
+    Nightly,
+}
+
+impl ExampleKind {
+    fn infer(demo: bool, nightly: bool) -> Self {
+        if demo && nightly {
+            Self::Nightly
+        } else if demo {
+            Self::Demo
+        } else {
+            Self::Ignore
+        }
+    }
+}
+
+#[derive(serde::Deserialize)]
+struct Frontmatter {
+    #[serde(default)]
+    title: String,
+    #[serde(default)]
+    tags: Vec<String>,
+    #[serde(default)]
+    description: String,
+    #[serde(default)]
+    thumbnail: String,
+    #[serde(default)]
+    thumbnail_dimensions: [u64; 2],
+    #[serde(default)]
+    demo: bool,
+    #[serde(default)]
+    nightly: bool,
+    #[serde(default)]
+    build_args: Vec<String>,
+}
+
+fn parse_frontmatter<P: AsRef<Path>>(path: P) -> anyhow::Result<Option<Frontmatter>> {
+    let path = path.as_ref();
+    let content = std::fs::read_to_string(path)?;
+    let content = content.replace('\r', ""); // Windows, god damn you
+    re_build_tools::rerun_if_changed(path);
+    let Some(content) = content.strip_prefix("---\n") else {
+        return Ok(None);
+    };
+    let Some(end) = content.find("---") else {
+        anyhow::bail!("{:?} has invalid frontmatter", path);
+    };
+    Ok(Some(serde_yaml::from_str(&content[..end]).map_err(
+        |e| {
+            anyhow::anyhow!(
+                "failed to read {:?}: {e}",
+                path.parent().unwrap().file_name().unwrap()
+            )
+        },
```

Comment:
We always call errors `err` in our code for consistency

```suggestion
        |err| {
            anyhow::anyhow!(
                "failed to read {:?}: {err}",
                path.parent().unwrap().file_name().unwrap()
            )
        },
```